jdk/src/share/classes/sun/security/provider/SunEntries.java

```diff
@@ -243,6 +247,16 @@ static void putEntries(Map<Object, Object> map) {
          * Set the digest provider based on whether native crypto is
          * enabled or not.
          */
+        /* Don't use native MD5 on AIX due to an observed performance regression. */
+        if (useNativeDigest
+            && NativeCrypto.isMD5Available()
+            && !isAIX
+        ) {
+            providerMD5 = "sun.security.provider.NativeMD5";
+        } else {
+            providerMD5 = "sun.security.provider.MD5";
```

Comment:
`providerMD5` needs to actually be used (i.e. on line 275).

Reply:
Sorry about that. Fixed.